python/lsst/daf/butler/core/datasets/type.py

```diff
@@ -273,6 +274,21 @@ def makeComponentDatasetType(self, component: str) -> DatasetType:
         return DatasetType(self.componentTypeName(component), dimensions=self.dimensions,
                            storageClass=self.storageClass.allComponents()[component])
 
+    def makeAllComponentDatasetTypes(self) -> List[DatasetType]:
+        """Return all the component dataset types assocaited with this
+        dataset type.
+
+        Returns
+        -------
+        all : `list` of `DatasetType`
+            All the component dataset types. If this is not a composite
+            then returns an empty list.
+        """
+        if not self.isComposite():
+            return []
```

Comment:
Do you actually need this check?  I'd have thought `storageClass.allComponents()` would return an empty iterable in this case anyway.

Reply:
Yes. I was prematurely optimizing.